lib/Dialect/FIRRTL/Transforms/ModuleInliner.cpp

```diff
@@ -729,6 +729,28 @@ void Inliner::cloneAndRename(
     newAnnotations.push_back(anno);
   }
 
+  // Handle inlining of RefType ops. Temporary wires of RefType must be removed
+  // here. The RefType wires were created to replace ports of inlined modules.
+  // RefType ports can only have out Direction. And there is only one connect to
+  // the output reftype port. Given these constraints, remove the connect, and
+  // directly replace the users with the connect.getSrc(). To ensure
+  // hierarchical inlining is correctly done, update any temporary RefType wires
+  // from the mapper.
```

Comment:
We're relaxing the downwards only restriction meaning that we can now see `RefType` ports with input direction.

Both the comment and the logic should be updated to handle this.